lib/PuppeteerSharp/ChromeTargetManager.cs

```diff
@@ -23,6 +23,7 @@ internal class ChromeTargetManager : ITargetManager
         private readonly ConcurrentDictionary<string, TargetInfo> _discoveredTargetsByTargetId = new();
         private readonly ConcurrentDictionary<ICDPConnection, List<TargetInterceptor>> _targetInterceptors = new();
         private readonly List<string> _targetsIdsForInit = new();
+        private readonly TaskCompletionSource<bool> _targetDiscoveryCompletionSource = new();
```

Comment:
Let's add a comment mentioning that this is .NET and we need this to avoid race conditions.

Reply:
Good question. As opposed to "uptream" (the original puppeteer implementation we follow).
It's important for us to follow which changes are not in upstream, because we don't want to remove our custom code.